twenty-first/src/util_types/algebraic_hasher.rs

```diff
@@ -68,34 +100,19 @@ pub trait AlgebraicHasher: SpongeHasher {
     }
 
     fn hash_varlen(input: &[BFieldElement]) -> Digest {
-        // calculate padded length
+        // calculate padded length; padding is at least one element
         let padded_length = roundup_nearest_multiple(input.len() + 1, RATE);
 
-        // pad input
+        // pad input with [1, 0, 0, ...]
         let input_iter = input.iter();
         let padding_iter = [&BFIELD_ONE].into_iter().chain(iter::repeat(&BFIELD_ZERO));
-        let padded_input = input_iter
-            .chain(padding_iter)
-            .take(padded_length)
-            .chunks(RATE);
-        let mut padded_input_iter = padded_input.into_iter().map(|chunk| {
-            chunk
-                .into_iter()
-                .copied()
-                .collect::<Vec<_>>()
-                .try_into()
-                .unwrap()
-        });
+        let padded_input = input_iter.chain(padding_iter).take(padded_length);
 
-        // absorb_init
-        let absorb_init_elems: [BFieldElement; RATE] =
-            padded_input_iter.next().expect("at least one absorb");
-        let mut sponge = Self::absorb_init(&absorb_init_elems);
+        // initialize sponge
```

Comment:
```suggestion
```

Reply:
This is parroting the code and thus seems quite redundant because it says basically exactly the same thing as the line below that, with which I'm trying to say that the meaning of the code is written down multiple times, _i.e._, more than once.